http://bugs.gentoo.org/show_bug.cgi?id=308735

These can be found in dosbox sf.net bugtracking system:
http://sourceforge.net/tracker/?group_id=52551&atid=467232

--- configure.in
+++ configure.in
@@ -5,7 +5,7 @@
 
 dnl Detect the canonical host and target build environment
 AC_CANONICAL_HOST
-AC_CANONICAL_TARGET
+AC_CANONICAL_BUILD
 
 dnl Setup for automake
 AM_INIT_AUTOMAKE
@@ -21,7 +21,7 @@
 
 dnl Some needed libaries for OS2
 dnl perharps join this with the other target depended checks. move them upwards
-if test x$target = xi386-pc-os2-emx ; then
+if test x$build = xi386-pc-os2-emx ; then
     CXXFLAGS="$CXXFLAGS -Zmt"
     LDFLAGS="$LDFLAGS -Zomf -Zmt"
     LIBS="$LIBS -los2me"
@@ -231,7 +231,7 @@
 dnl The target cpu checks for dynamic cores
 AH_TEMPLATE(C_TARGETCPU,[The type of cpu this target has])
 AC_MSG_CHECKING(for target cpu type) 
-case "$target_cpu" in
+case "$build_cpu" in
   x86_64 | amd64)
     AC_DEFINE(C_TARGETCPU,X86_64)
     AC_MSG_RESULT(x86-64 bit compatible)
@@ -343,7 +343,7 @@
 
 AH_TEMPLATE(C_SSHOT,[Define to 1 to enable screenshots, requires libpng])
 AC_CHECK_HEADER(png.h,have_png_h=yes,)
-AC_CHECK_LIB(png, png_check_sig, have_png_lib=yes, ,-lz)
+AC_CHECK_LIB(png, png_get_io_ptr, have_png_lib=yes)
 if test x$have_png_lib = xyes -a x$have_png_h = xyes ; then
   LIBS="$LIBS -lpng -lz"
   AC_DEFINE(C_SSHOT,1)
@@ -355,7 +355,7 @@
 AH_TEMPLATE(C_IPX,[Define to 1 to enable IPX over Internet networking, requires SDL_net])
 AC_CHECK_HEADER(SDL_net.h,have_sdl_net_h=yes,)
 
-if test x$target = xi386-pc-os2-emx ; then
+if test x$build = xi386-pc-os2-emx ; then
   AC_MSG_CHECKING(for SDLNet_Init in SDL_net);
   LIBS_BACKUP=$LIBS;
   LIBS="$LIBS -lSDL_Net";
@@ -393,7 +393,7 @@
 AC_ARG_ENABLE(opengl,AC_HELP_STRING([--disable-opengl],[Disable opengl support]),,enable_opengl=yes)
 AC_MSG_CHECKING(whether opengl display output will be enabled) 
 if test x$enable_opengl = xyes; then
-case "$target" in
+case "$build" in
     *-*-darwin*)
        AC_MSG_RESULT(yes)
        LIBS="$LIBS -framework OpenGL"
@@ -448,7 +448,7 @@
 
 
 dnl Some target detection and actions for them
-case "$target" in
+case "$build" in
     *-*-cygwin* | *-*-mingw32*)
        LIBS="$LIBS -lwinmm"
        AC_CHECK_HEADERS(ddraw.h)
@@ -486,7 +486,7 @@
 esac
 
 dnl Some stuff for the icon.
-case "$target" in
+case "$build" in
     *-*-cygwin* | *-*-mingw32*)
        dnl Some stuff for the ico
        AC_CHECK_TOOL(WINDRES, windres, :)
